src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -173,7 +173,14 @@ protected override void Dispose(bool disposing)
                         var attr = File.GetAttributes(copiedFile);
                         if (!attr.HasFlag(FileAttributes.Directory))
                         {
-                            File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            try
+                            {
+                                File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            }
+                            catch (ArgumentOutOfRangeException)
```

Comment:
Never mind that, the problem doesn't involve dates in the future. The DateTime for some zip file entries is just not set, and those are the ones that hit this bug.